pkg/config/loader_test.go

```diff
@@ -0,0 +1,305 @@
+package config
+
+import (
+	"os"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+
+	autoheal "github.com/openshift/autoheal/pkg/apis/autoheal/v1alpha2"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestFile(t *testing.T) {
+	l := NewLoader()
+	l.File("test_file")
+	if len(l.files) != 1 {
+		t.Fail()
+	}
+
+	if l.files[0] != "test_file" {
+		t.Fail()
+	}
```

Comment:
I think it would be better if we don't depend on the private `files` field, otherwise this test will stop working if we change those internals. I'd suggest to either remove this kind of test, or else rewrite it in such a way that it only uses the public interface of the `Loader` type.